lib/aggregators/base-aggregator.js

```diff
@@ -9,18 +9,32 @@ const EventEmitter = require('events').EventEmitter
 const logger = require('../logger').child({ component: 'base_aggregator' })
 
 class Aggregator extends EventEmitter {
-  constructor(opts, collector) {
+  constructor(opts, collector, harvester) {
     super()
 
     this.defaultPeriod = this.periodMs = opts.periodMs
     this.defaultLimit = this.limit = opts.limit
     this.runId = opts.runId
     this.isAsync = opts.isAsync || false
+    // function to pass in to determine if we can start a given aggregator
+    this.isEnabled =
+      opts.enabled ||
+      function defaultEnabled() {
+        return true
+      }
+    this.enabled = this.isEnabled(opts.config)
```

Comment:
I've been reading this over and over for a minute or two. It's giving me some sort of "uhhh" feeling. Is this just some backward compatibility thing?

Reply:
so an aggregator can be enabled but based on server side configuration updates become disabled.  This centralizes the configuration bits to determine that.  

This used to live [here](https://github.com/newrelic/node-newrelic/blob/main/lib/agent.js#L310), [here](https://github.com/newrelic/node-newrelic/blob/main/lib/agent.js#L451), and [here](https://github.com/newrelic/node-newrelic/blob/main/lib/agent.js#L485).  It has to be a function to evaluate at the config can change in reconfigure, I'm totally open for better solutions, it was the best i could think of at the time